### PR TITLE
Assorted tweaks to careers app.

### PR DESCRIPTION
**Pull request by lewiscollard**
* Add an expiry date field, obey it in views, allow filtering active/non-active in the admin.
* Add an oft-used off-site application URL field.
* Allow application email to be empty.
* Inherit detail view from PageDetailMixin so that the title (and SEO stuff) is obeyed.
* Add a 'get some careers' template tag.
* Remove non-functionining PageBaseAdmin.NAVIGATION_FIELDS from career admin.
* General code style tweaks

### CODE REVIEW

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/careers/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from cms.apps.pages.middleware import RequestPageManager
+from django.test import RequestFactory
+
+from ..views import CareerDetailView, CareerListView
+from ._base import CareersBaseTestCase
+
+
+class CareerViewsTestCase(CareersBaseTestCase):
+    # Remeber _base gives us:
```
Comment (danielsamuels) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/careers/tests/test_views.py]: Remember*

Comment (lewiscollard) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/careers/tests/test_views.py]: Thanks, fixed :)

**Comment by dan-gamble** [comment]
Is this one good to go in or changes still needed?

**Review by dan-gamble — APPROVED** [review]

**Comment by lewiscollard** [comment]
GTG in my opinion :)

**Pull request merged**